compiler/rustc_query_system/src/dep_graph/serialized.rs

```diff
@@ -225,7 +262,8 @@ impl<K: DepKind> EncoderState<K> {
         }
         debug!(?position);
 
-        let index = self.nodes.push((node.node, position.try_into().unwrap()));
+        let index = self.positions.push(position.try_into().unwrap());
+        self.index.insert(&node.node, &SerializedDepNodeIndex::from_u32(index.as_u32()));
```

Comment:
It might make sense to move building the odht to `finish()`, so that we don't have to keep growing it.